### PR TITLE
Add `MergeService` to make the bot run

### PR DESCRIPTION
**Pull request by dmcrodrigues**
Ticket: https://babylonpartners.atlassian.net/browse/CNSMR-2254

## Why?

We are in the process of moving our existent bot to be this repository to make it OSS and continue development with further improvements and new features.

## How?

This keeps the same existent infrastructure and brings the existent models except for some namespaces what were removed. `PullRequestContext` was renamed to `PullRequestMetadata` as this model was already added here. Other than that everything was kept the same and only the following PRs will start to introduce changes.

## What's missing?

~I still need to implement the API layer to integrate with GitHub and then pipe the events to the service to make it work. I'm hoping to finish this very soon.~

Due the significant number of changes that this will require, it's better to have them in a separate PR.

### CODE REVIEW

File: Sources/Bot/Services/MergeService.swift [hunk 1]
```diff
@@ -0,0 +1,615 @@
...
+    private let statusChecksCompletion: Signal<StatusChange, NoError>
+    private let statusChecksCompletionObserver: Signal<StatusChange, NoError>.Observer
+
+    public init(
+        integrationLabel: String,
+        statusChecksTimeout: TimeInterval = 60.minutes,
+        logger: LoggerProtocol,
+        github: GitHubAPIProtocol,
+        scheduler: DateScheduler = QueueScheduler()
+        ) {
```
Comment (karabatov) [inline Sources/Bot/Services/MergeService.swift]:
```suggestion
    ) {
```
🤓

Comment (karabatov) [reply Sources/Bot/Services/MergeService.swift]: Overall the indentation is weird in places :) Clearly the result of a copy.

Comment (dmcrodrigues) [reply Sources/Bot/Services/MergeService.swift]: I'll make a pass to review everything @karabatov, thanks!

**Comment by dmcrodrigues** [comment]
@ilyapuchka @karabatov I fixed the indentation and I think it's better to merge this one as is since implementing the GitHub API will bring some changes that should be reviewed isolated.

**Review by karabatov — APPROVED** [review]

**Review by ilyapuchka — APPROVED** [review]

**Pull request merged**